### PR TITLE
Benchmark refactor

### PR DESCRIPTION
**Pull request by rafael1193**
Right now, the only way to launch the planning algorithm is through the benchmark script, that is getting quite large.  I moved to other files most of the functionality to launch the VNS planning and to display the results. This way, interface and logic are kept separate.

Merging this PR is required for using SAOP in other contexts than benchmarking. In the near future, we should be able to re-plan and to communicate with a UAV ground control station in order to "close the loop"

Note: The command line interface of benchmark.py is the same as before. This PR only reorders what is inside.

### CODE REVIEW

File: cpp/src/vns/plan.hpp [hunk 1]
```diff
@@ -60,12 +62,16 @@ struct Plan {
                     Cell c{x, y};
                     possible_observations.push_back(
                             //FIXME: This shouldn't be 3D
```
Comment (arbimo) [inline cpp/src/vns/plan.hpp]: Remove fixme that is fixed

File: cpp/src/vns/plan.hpp [hunk 2]
```diff
@@ -42,12 +42,14 @@ struct Plan {
     TimeWindow time_window;
     Trajectories core;
     shared_ptr<FireData> firedata;
-    vector<Point3dTimeWindow> possible_observations;
+    vector<PointTimeWindow> possible_observations;
+    vector<PointTimeWindow> actual_observations;
```
Comment (arbimo) [inline cpp/src/vns/plan.hpp]: What is the purpose of actual_observations ?
I don't see it anywhere except on initialization.

The reason I did not include previously is that keeping track of actual observations must be made incrementally which is quickly error prone and would not probably not buy much computational cost.

**Review by arbimo — APPROVED** [review]
Good idea to do this.

I have not tested the actual changes be they seem fine and definitely are an improvement over the previous state of affair.

File: cpp/src/vns/plan.hpp (hunk 2 above)
Comment (rafael1193) [reply cpp/src/vns/plan.hpp]: It is just a bogus declaration that souldn't be there. I was wondering how to initialize plans when replanning. I wrote it, but i'm not using it, that's why you don't see this anywhere.

You're right on that, keeping track of this is quite error prone.

File: cpp/src/vns/plan.hpp (hunk 1 above)
Comment (rafael1193) [reply cpp/src/vns/plan.hpp]: Done

**Comment by arbimo** [comment]
Looks good to me.

Maybe a last thing we could do before (or after) merging is remove all VNS configuration from `benchmark.py`, and provide a external file with some VNS configurations that would be used by default.

**Comment by rafael1193** [comment]
I removed all the VNS confs as you said. I just kept one simple one, called "demo", just to keep benchmark.py able to run by itself.

**Pull request merged**